newIDE/app/src/Profile/EmailVerificationPendingDialog.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import React from 'react';
+import FlatButton from '../UI/FlatButton';
+import RaisedButton from '../UI/RaisedButton';
+import Dialog from '../UI/Dialog';
+import { type AuthenticatedUser } from './AuthenticatedUserContext';
+import { Column, Line, Spacer } from '../UI/Grid';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import BackgroundText from '../UI/BackgroundText';
+import VerifiedUser from '@material-ui/icons/VerifiedUser';
+import Text from '../UI/Text';
+import { useInterval } from '../Utils/UseInterval';
+
+type Props = {|
+  onClose: Function,
+  authenticatedUser: AuthenticatedUser,
+|};
+
+export default function EmailVerificationPendingDialog({
+  onClose,
+  authenticatedUser,
+}: Props) {
+  const isVerified =
+    !!authenticatedUser &&
+    !!authenticatedUser.firebaseUser &&
+    !!authenticatedUser.firebaseUser.emailVerified;
+  useInterval(
+    () => authenticatedUser.onRefreshFirebaseProfile(),
+    isVerified ? null : 3900
+  );
+
+  return (
+    <Dialog
+      actions={[
+        isVerified ? (
+          <RaisedButton
+            label={<Trans>Done!</Trans>}
+            key="close"
+            primary
+            onClick={onClose}
+          />
+        ) : (
+          <FlatButton
+            label={<Trans>Cancel and close</Trans>}
+            key="close"
+            primary={false}
+            onClick={onClose}
+          />
+        ),
+      ]}
+      title={undefined}
+      maxWidth="sm"
+      cannotBeDismissed={true}
+      open
+      noMargin
+    >
+      {!isVerified && (
+        <Column>
+          <Line justifyContent="center" alignItems="center">
+            <CircularProgress size={20} />
+            <Spacer />
+            <Text>Waiting for the email verification...</Text>
+          </Line>
+          <Spacer />
+          <Line justifyContent="center">
+            <BackgroundText>
+              <Trans>
+                Check your inbox and click the link to verify your email address
+                - and come back here when it's done.
+              </Trans>
+            </BackgroundText>
+          </Line>
+        </Column>
+      )}
+      {isVerified && (
+        <Column>
+          <Line>
+            <Text>
+              <Trans>Your email is now verified!</Trans>
+            </Text>
+            <Spacer />
+            <VerifiedUser />
+          </Line>
```

Comment:
Design speaking: look at the alignment of the icon:
![image](https://user-images.githubusercontent.com/1280130/135998435-643ed7e8-7517-4abd-abcf-a7cb0634a75a.png)

We probably want to align it vertically. We also have the loading that was horizontally centered, so we might want to do the same thing:
![image](https://user-images.githubusercontent.com/1280130/135998561-e42292db-edad-4517-b4a8-ffc54623e980.png)

Something like?

```suggestion
          <Line justifyContent="center" alignItems="center">
            <VerifiedUser />
            <Spacer />
            <Text>
              <Trans>Your email is now verified!</Trans>
            </Text>
          </Line>
```

(also note that this kind of design nits are easier to refactor/fix later if we have a storybook story :))